apps/clientside-bot/test/utils.ts

```diff
@@ -0,0 +1,237 @@
+import {
+	Category,
+	Community,
+	GuildConfig,
+	Report,
+	Revocation,
+} from "fagc-api-types"
+import faker from "faker"
+import FAGCBan from "../src/database/FAGCBan"
+import PrivateBan from "../src/database/PrivateBan"
+import Whitelist from "../src/database/Whitelist"
+
+export function createTimes<F extends (...args: any[]) => any>(
+	creator: F,
+	params: Parameters<F> | (() => Parameters<F>),
+	count: number
+): ReturnType<F>[]
+export function createTimes<F extends () => any>(
+	creator: F,
+	count: number
+): ReturnType<F>[]
+export function createTimes<F extends (...args: any[]) => any>(
+	creator: F,
+	params: Parameters<F> | (() => Parameters<F>) | number,
+	count?: number
+): ReturnType<F>[] {
+	if (typeof params === "number") {
+		return Array.from({ length: params }, () => creator())
+	}
+
+	// TODO: remove the type cast when i figure out why `count` can be undefined here according to types - which it can't
+	return Array.from({ length: count as number }, () =>
+		typeof params === "function" ? creator(...params()) : creator(...params)
+	)
+}
+
+// this is to make sure that the IDs are unique
+const existingFAGCIds = new Set<string>()
+export const createFAGCId = (): string => {
+	const id = faker.random.alphaNumeric(6)
+	if (existingFAGCIds.has(id)) return createFAGCId()
+	existingFAGCIds.add(id)
+	return id
+}
+
+/**
+ * Get a random element from the provided array
+ */
+export const randomElementFromArray = <T>(array: T[]): T => {
+	return array[Math.floor(Math.random() * array.length)]
+}
+
+export const randomElementsFromArray = <T>(array: T[], count?: number): T[] => {
+	if (!count) {
+		count = Math.floor(Math.random() * array.length) + 1
+	}
```

Comment:
Would this function ever be called with an undefined count when the overload above covers this case?

Reply:
They are two distinct functions, `Element` and `Elements`.